bundles/automation/org.eclipse.smarthome.automation.module.script/src/main/java/org/eclipse/smarthome/automation/module/script/internal/handler/AbstractScriptModuleHandler.java

```diff
@@ -106,10 +107,13 @@ private void loadConfig() {
     protected void setExecutionContext(ScriptEngine engine, Map<String, ?> context) {
         ScriptContext executionContext = engine.getContext();
 
-        // make the whole context available as "ctx"
+        // Add the rule's UID to the context
+        final Map<String, Object> contextNew = new HashMap<>(context);
+        contextNew.put("ruleUID", this.ruleUID);
+
+        // make the whole (new) context available as "ctx"
         // Note: We don't use "context" here as it doesn't work on all JVM versions!
-        executionContext.setAttribute("ctx", context, ScriptContext.ENGINE_SCOPE);
-        executionContext.setAttribute("ruleUID", this.ruleUID, ScriptContext.ENGINE_SCOPE);
```

Comment:
Looks like we are back at the same point where also  https://github.com/eclipse/smarthome/pull/3152#issuecomment-287660146 got stuck. If we ever want to have a chance to offer a versioned "API" to scripts, then the global namespace would be poisonous in general. But the ruleUID in particular is so elemental that we can safely accept the "risk" of having it globally as well, IMHO.